pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -328,12 +336,36 @@ def tpu_id(self) -> Optional[int]:
     def has_ipu(self) -> bool:
         # Here, we are not checking for IPU availability, but instead if User has passed
         # `ipus` to Trainer for training.
-        return self.ipus is not None or isinstance(self._training_type_plugin, IPUPlugin)
+        if self.ipus is not None or isinstance(self._training_type_plugin, IPUPlugin):
+            return True
+        return self._map_devices_to_accelerator(DeviceType.IPU)
 
     @property
     def use_ipu(self) -> bool:
         return self._accelerator_type == DeviceType.IPU and self.has_ipu
 
+    def _set_devices_to_cpu_num_processes(self) -> None:
+        if self.num_processes <= 1:
+            self._map_devices_to_accelerator(DeviceType.CPU)
+
+    def _map_devices_to_accelerator(self, accelerator: str) -> bool:
+        if self.devices is None:
+            return False
+        if accelerator == DeviceType.TPU and _TPU_AVAILABLE:
+            self.tpu_cores = device_parser.parse_tpu_cores(self.devices)
+            return True
+        elif accelerator == DeviceType.IPU and _IPU_AVAILABLE:
+            self.ipus = self.devices
+            return True
+        elif accelerator == DeviceType.GPU and torch.cuda.is_available():
+            self.gpus = self.devices
```

Comment:
Should we set num_processes to devices there too ?

Reply:
Yes, we do set `num_processes` to `devices` for `cpu` acclerator.

Reply:
what is the usage of `num_processes` when you run GPU, is there any?

Reply:
No, it doesn't have an impact.